OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -221,10 +225,32 @@ public virtual Barrel CheckFire(Actor self, IFacing facing, Target target)
 			if (!Weapon.IsValidAgainst(target, self.World, self))
 				return null;
 
+			if (ticksSinceLastShot >= Weapon.ReloadDelay)
+				Burst = Weapon.Burst;
+
+			ticksSinceLastShot = 0;
+
 			var barrel = Barrels[Burst % Barrels.Length];
 			Func<WPos> muzzlePosition = () => self.CenterPosition + MuzzleOffset(self, barrel);
 			var legacyFacing = MuzzleOrientation(self, barrel).Yaw.Angle / 4;
 
+			var passiveTarget = Weapon.TargetActorCenter ? target.CenterPosition : target.Positions.PositionClosestTo(muzzlePosition());
+			var initialOffset = Weapon.FirstBurstTargetOffset;
+			if (initialOffset != WVec.Zero)
+			{
+				// We want this to match Armament.LocalOffset, so we need to convert it to forward, right, up
+				initialOffset = new WVec(initialOffset.Y, -initialOffset.X, initialOffset.Z);
+				passiveTarget += initialOffset.Rotate(WRot.FromFacing(legacyFacing));
```

Comment:
If you're fine with it, I'd actually like to leave it like this. If I or any other modder ever needs a different behavior, we can still fix it (or make it configurable).